src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDiagnosticsTranslator.cs

```diff
@@ -29,7 +30,17 @@ internal class DefaultLSPDiagnosticsTranslator : LSPDiagnosticsTranslator
         {
             _documentManager = documentManager;
             _requestInvoker = requestInvoker;
-            _logger = loggerProvider.CreateLogger(nameof(DefaultLSPDiagnosticsTranslator));
```

Comment:
I don't understand how this was previously an issue? Logger provider isn't lazy, nor an optional import. Was it a problem in the actual loggerProvider implementation?

Reply:
+1, I don't understand this part :/

Reply:
The problem was that https://github.com/dotnet/razor-tooling/blob/670541ebcdc44d9de7941368e7f128c983e9f8fc/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/HTMLCSharpLanguageServerLogHubLoggerProvider.cs#L73 needs to be called before CreateLogger is (to create the underlying _loggerProvider). As-is we have no guarantee of that, we just rely on timing to get `InitializeLoggerAsync` called first. By creating the logger on-demand (from an async method) we can avoid that ambiguity.

My preferred fix would be to do away with `InitializeLoggerAsync` all-together (I don't care for the `Initialize` pattern), but since we're implementing `ILoggerProvider` I didn't see a way to do that (since CreateLogger is sync). If anybody has alternate suggestions (or wants to talk more about context to generate some) I'm down.

Reply:
Ahh okay. I missed that this was a logger provider, not a logger factory, so wasn't thinking that they worked differently. Definitely not a great design, but I don't think its worth blocking this PR on or anything.

Reply:
Ya I'm not a fan of the pattern either 😢. Once we're normalized to a single language server in the language server re-design this should go away at least 😄